sdk/appcenter/src/main/java/com/microsoft/appcenter/http/HttpClientRetryer.java

```diff
@@ -91,8 +93,20 @@ public synchronized void cancel() {
         @Override
         public void onCallFailed(Exception e) {
             if (mRetryCount < RETRY_INTERVALS.length && HttpUtils.isRecoverableError(e)) {
-                long delay = RETRY_INTERVALS[mRetryCount++] / 2;
-                delay += mRandom.nextInt((int) delay);
+                long delay = 0;
+
+                if (e instanceof HttpException) {
+                    HttpException httpException = (HttpException) e;
+                    if (httpException.getHeaders().containsKey(RETRY_AFTER_MS_KEY)) {
+                        delay = Long.parseLong(httpException.getHeaders().get(RETRY_AFTER_MS_KEY));
+                    }
+                }
+
+                if(delay == 0){
```

Comment:
Reformat file, please always commit via Android Studio version control window and use the following options:

<img width="282" alt="screen shot 2019-02-28 at 4 51 14 pm" src="https://user-images.githubusercontent.com/623672/53661203-aa8eab80-3c14-11e9-8bdb-69a80129a759.png">